# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/EncodedVideoChunk.h
# Commit: 226dda36eb9e
# Full Hash: 226dda36eb9eefe20a2d9ca5122fbcd5d8805a94
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Implement VideoDecoder::Decode r=padenot
#   
#   This patch implements the decode method [1] for WebCodecs' VideoDecoder
#   interface, and add some code that reset method needs.
#   
# ==============================================================================

diff -r 43451f9834e7 -r 226dda36eb9e dom/media/webcodecs/EncodedVideoChunk.h
--- a/dom/media/webcodecs/EncodedVideoChunk.h	Thu Jun 29 22:33:12 2023 +0000
+++ b/dom/media/webcodecs/EncodedVideoChunk.h	Thu Jun 29 22:33:12 2023 +0000
@@ -67,6 +67,9 @@
       const MaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer& aDestination,
       ErrorResult& aRv);
 
+  // Non-webidl method.
+  uint8_t* Data() { return mBuffer.get(); }
+
  private:
   // EncodedVideoChunk can run on either main thread or worker thread.
   void AssertIsOnOwningThread() const {